client/modules/workspace/src/JobsDetailModal/JobsDetailModal.tsx

```diff
@@ -1,39 +1,144 @@
+import { formatDateTime } from '../utils/timeFormat';
 import { TJob } from '@client/hooks';
 
+const DataFilesLink: React.FC<
+  React.PropsWithChildren<{
+    to: string;
+  }>
+> = ({ to, children }) => {
+  return (
+    <NavLink to={to} className={styles.link}>
+      {children}
+    </NavLink>
+  );
+};
+
 export const JobsDetailModalBody: React.FC<{
   isOpen: boolean;
   uuid: string;
-}> = ({ isOpen, uuid }) => {
+  onToggleModal: (isOpen: boolean) => void;
+}> = ({ isOpen, uuid, onToggleModal }) => {
   const { data, isLoading } = useGetJobs('select', { uuid }) as {
     data: TJob;
     isLoading: boolean;
   };
+  const [isModalOpen, setIsModalOpen] = useState(isOpen);
+
+  useEffect(() => {
+    setIsModalOpen(isOpen);
+  }, [isOpen]);
+
+  const handleCancel = () => {
+    setIsModalOpen(false);
+    onToggleModal(false);
+  };
 
   return (
     <Modal
-      title={<h2>Job Detail: {uuid}</h2>}
+      className={`${styles.root} job-history-modal`}
+      title={
+        data && (
+          <>
+            <header>
+              Job Detail: {data.name}-{uuid}
+              <dl className={styles['header-details']}>
+                <dt>Job UUID: </dt>
+                <dd>{uuid}</dd>
+                <dt>Application: </dt>
+                <dd>{data.name}</dd>
+                <dt>System: </dt>
+                <dd>{data.execSystemId}</dd>
+              </dl>
+            </header>
+          </>
+        )
+      }
       width="60%"
-      open={isOpen}
+      open={isModalOpen}
+      onCancel={handleCancel}
       footer={null} // Remove the footer from here
     >
-      <div className={styles.modalContentContainer}>
+      <div className={styles['modal-body-container']}>
         {isLoading && <Spin className={styles.spinner} />}
-        {data && isOpen && <span>{data.name}</span>}
+        {data && isOpen && (
+          <>
+            <div className={`${styles['left-panel']}`}>
+              <dl>
+                <dt>Execution:</dt>
+                <dd>
+                  <DataFilesLink
+                    to={`./data/browser/${data.archiveSystemId}${data.archiveSystemDir}`}
+                  >
+                    View in Data Files
+                  </DataFilesLink>
+                </dd>
+                <dt>Output:</dt>
+                <dd>
+                  <DataFilesLink
+                    to={`./data/browser/${data.archiveSystemId}${data.archiveSystemDir}`}
+                  >
+                    View in Data Files
+                  </DataFilesLink>
+                </dd>
+              </dl>
+              <Button type="primary" className={styles['submit-button']}>
+                Resubmit Job
+              </Button>
+              <Button type="primary" className={styles['submit-button']}>
+                Cancel Job
+              </Button>
+
+              <Button type="primary" className={styles['submit-button']} danger>
+                Delete Job
+              </Button>
```

Comment:
We'll want to conditionally render these based on the job status. e.g. if the job is finished, cancel job should not be an option